pkg/reconciler/reconciler_test.go

```diff
@@ -738,8 +738,15 @@ var _ = Describe("Reconciler", func() {
 									Patch: func(ctx context.Context, _ client.WithWatch, fakeObj client.Object, patch client.Patch, opts ...client.PatchOption) error {
 										return mgr.GetClient().Patch(ctx, fakeObj, patch, opts...)
 									},
-									SubResource: func(_ client.WithWatch, subresource string) client.SubResourceClient {
-										return mgr.GetClient().SubResource(subresource)
+									SubResourceUpdate: func(ctx context.Context, client client.Client, subResourceName string, obj client.Object, opts ...client.SubResourceUpdateOption) error {
+										// workaround https://github.com/kubernetes/kubernetes/issues/124347
+										err := mgr.GetClient().SubResource(subResourceName).Update(ctx, obj, opts...)
+										if err != nil {
+											if apierrors.IsConflict(err) {
+												return apierrors.NewNotFound(gvk.GroupVersion().WithResource("testapps").GroupResource(), obj.GetName())
```

Comment:
I think the comment is about the error conversion, so how about putting it there?
```suggestion
										err := mgr.GetClient().SubResource(subResourceName).Update(ctx, obj, opts...)
										if err != nil {
											if apierrors.IsConflict(err) {
												// workaround https://github.com/kubernetes/kubernetes/issues/124347
												return apierrors.NewNotFound(gvk.GroupVersion().WithResource("testapps").GroupResource(), obj.GetName())
```